lib/internal/streams/add-abort-signal.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const {
+  AbortError,
+  codes,
+} = require('internal/errors');
+
+const { ERR_INVALID_ARG_TYPE } = codes;
+
+const {
+  validateAbortSignal,
+} = require('internal/validators');
+
+function isReadable(obj) {
+  return !!(obj && typeof obj.pipe === 'function');
+}
+
+module.exports = function addAbortSignal(signal, readable) {
```

Comment:
I would prefer to have the args in the reverse order.

Reply:
Why?

Reply:
Usually the object that a function applies on/mutates is the first argument... which is how we usually do it. I think?

I guess as last argument I guess also makes sense in functional style with currying.

No strong opinion.

Reply:
Yeah this is so I can `addAbortSignal.bind(null, signal)` and run it over multiple streams but that's kind of a stretch and I have no real strong feelings either.